bin/metrics-tomcat-jmxproxy.rb

```diff
@@ -0,0 +1,132 @@
+#! /usr/bin/env ruby
+#
+#   metrics-tomcat-jmxproxy
+#
+# DESCRIPTION:
+#   Return value of JMX attribute to Graphite via Tomcat's JMXProxy
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: unirest
+#
+# USAGE:
+#  Basic Usage:
+#    ./metrics-tomcat-jmxproxy.rb
+#      --url http://localhost:8080
+#      --account admin
+#      --password password
+#      --bean 'java.lang:type=MemoryPool,name=Compressed Class Space'
+#      --attribute PeakUsage
+#      --key used
+#      --scheme my.scheme
+#
+#    my.scheme.PeakUsage.used 2523848 1513887865
+#
+# NOTES:
+#
+# See http://localhost:8080/docs/manager-howto.html for security information
+
+require 'sensu-plugin/metric/cli'
+require 'unirest'
+require 'uri'
+
+class MetricsTomcatJMXProxy < Sensu::Plugin::Metric::CLI::Graphite
+  option :url,
+         description: 'URL of the Tomcat Manager',
+         short: '-u URL',
+         long: '--url URL',
+         default: 'http://localhost:8080',
+         required: true
+
+  option :account,
+         description: 'Username of the Tomcat User that has the manager-jmx role',
+         short: '-a USERNAME',
+         long: '--account USERNAME',
+         default: 'admin',
+         required: true
+
+  option :password,
+         description: 'Tomcat User password',
+         short: '-p PASSWORD',
+         long: '--password PASSWORD',
+         default: 'password',
```

Comment:
No, it's not the default, fortunately! The default is no users enabled so for jmxproxy to work, you have to add a user with the role `manager-jmx` to tomcat-users.xml. Also, [Tomcat has a valve to turn off access](https://tomcat.apache.org/tomcat-8.5-doc/manager-howto.html#Introduction) to things like manager, specifically.

BTW, if you're accustomed to turning off management, you really should check out what's available for monitoring. `jmxproxy` is nice because it is just plaintext.
https://wiki.apache.org/tomcat/FAQ/Monitoring
https://tomcat.apache.org/tomcat-8.5-doc/manager-howto.html#Using_the_JMX_Proxy_Servlet

Reply:
if its not the default, I see no value in specifying this default. let's remove this.